app/models/setting.rb

```diff
@@ -85,61 +68,83 @@ class Setting < ApplicationRecord
                  EUC-KR
                  Big5
                  Big5-HKSCS
-                 TIS-620)
+                 TIS-620).freeze
 
-  cattr_accessor :available_settings
+  class << self
+    def create_setting(name, value = {})
+      ::Settings::Definition.add(name, **value.symbolize_keys)
+    end
 
-  def self.create_setting(name, value = nil)
-    @@available_settings[name] = value
-  end
+    def create_setting_accessors(name)
+      return if [:installation_uuid].include?(name.to_sym)
 
-  def self.create_setting_accessors(name)
-    return if [:installation_uuid].include?(name.to_sym)
+      # Defines getter and setter for each setting
+      # Then setting values can be read using: Setting.some_setting_name
+      # or set using Setting.some_setting_name = "some value"
+      src = <<-END_SRC
+        def self.#{name}
+          # when running too early, there is no settings table. do nothing
+          self[:#{name}] if settings_table_exists_yet?
+        end
 
-    # Defines getter and setter for each setting
-    # Then setting values can be read using: Setting.some_setting_name
-    # or set using Setting.some_setting_name = "some value"
-    src = <<-END_SRC
-      def self.#{name}
-        # when running too early, there is no settings table. do nothing
-        self[:#{name}] if settings_table_exists_yet?
-      end
+        def self.#{name}?
+          # when running too early, there is no settings table. do nothing
+          return unless settings_table_exists_yet?
+          value = self[:#{name}]
+          ActiveRecord::Type::Boolean.new.cast(value)
```

Comment:
There is a minor bug here, taking `app_title` as an example:

```
Setting.app_title
=> "OpenProject"
Setting.app_title?
=> true
Setting.app_title = nil
=> nil
Setting.app_title?
=> false
Setting.app_title = 'f'
=> false
Setting.app_title?
=> false # should be true
```

It's contrived, I admit.

Reply:
I deprecated accessing `.[setting]?` for non boolean settings. Initially I planned to remove the functionality for those settings altogether but there are a couple of settings/configurations which rely on the existing functionality.

Once the method is limited the way it should be, the example will no longer apply.